vectordraw.py

```diff
@@ -48,7 +70,41 @@ def check_tip_x(check, vectors):
 def check_tip_y(check, vectors):
     vec = vectors[check['vector']]
     if _check_coordinate(check, vec.tip.y):
-        return 'Vector {} does not tip at correct point.'.format(vec.name)
+        return 'Vector {} does not end at correct point.'.format(vec.name)
+
+def _check_coords(vec, expected, tolerance):
+    for expected_coords, vec_coords in ([expected[0], vec.tail], [expected[1], vec.tip]):
+        if expected_coords[0] == '_':
+            if expected_coords[1] != '_':
+                # Only y given; not interested in x.
+                if abs(expected_coords[1] - vec_coords.y) > tolerance:
+                    return True
+        elif expected_coords[1] == '_':
+            # Only x given; not interested in y.
+            if abs(expected_coords[0] - vec_coords.x) > tolerance:
+                return True
```

Comment:
I'm confused by `False` meaning "correct" and `True` meaning "not correct" as well.

Yet another option how to implement this more elegantly:

    def _coord_delta(expected, actual):
        if expected == '_':
            return 0.0
        return expected - actual

    def _check_coords(vec, expected, tolerance):
        for expected_coords, vec_coords in ([expected[0], vec.tail], [expected[1], vec.tip]):
            delta_x = _coord_delta(expected_coords[0], vec_coords.x)
            delta_y = _coord_delta(expected_coords[1], vec_coords.y)
            if math.hypot(delta_x, delta_y) > tolerance:
                return True
        return False

Basically, set the error to zero for dimensions we don't care about.